packages/core-p2p/__tests__/peer-verifier.test.ts

```diff
@@ -807,4 +807,349 @@ describe("PeerVerifier", () => {
+                    height: 18,
+                    forgingAllowed: false,
+                    currentSlot: 18,
+                    header: {
+                        height: 19,
+                        id: "13965046748333390338",
+                    },
+                };
+
+                expect(await peerVerifier.checkState(claimedState, Date.now() + 2000)).toBeUndefined();
+            });
+        });
+
+        describe("when Case1. Peer height > our height and our highest block is part of the peer's chain", () => {
+            it("should return PeerVerificationResult forked", async () => {
+                const generatorPublicKey = "03c5282b639d0e8f94cfac6c0ed242d1634d8a2c93cbd76c6ed2856a9f19cf6a13";
+                const claimedState: Contracts.P2P.PeerState = {
+                    height: 18,
+                    forgingAllowed: false,
+                    currentSlot: 18,
+                    header: {
+                        height: 18,
+                        id: "13965046748333390338",
+                        generatorPublicKey,
+                    },
+                };
+                const ourHeader = {
+                    height: 15,
+                    id: "11165046748333390338",
+                };
+
+                stateStore.getLastHeight = jest.fn().mockReturnValue(ourHeader.height);
+                stateStore.getLastBlocks = jest
+                    .fn()
+                    .mockReturnValue([{ data: { height: ourHeader.height }, getHeader: () => ourHeader }]);
+                databaseInterceptor.getBlocksByHeight = jest.fn().mockImplementation((blockHeights) =>
+                    blockHeights.map((height: number) => ({
+                        height,
+                        id: height.toString().padStart(2, "0").repeat(20), // just using height to mock the id
+                    })),
+                );
+                peerCommunicator.hasCommonBlocks = jest.fn().mockImplementation((_, ids) => ({
+                    id: ids[ids.length - 1],
+                    height: parseInt(ids[ids.length - 1].slice(0, 2)),
+                }));
+                trigger.call = jest.fn().mockReturnValue([
+                    {
+                        getPublicKey: () => {
+                            return generatorPublicKey;
+                        },
+                    },
+                ]);
+                const spyFromData = jest
+                    .spyOn(Blocks.BlockFactory, "fromData")
+                    .mockImplementation(blockWithIdFromDataMock);
+
+                const result = await peerVerifier.checkStateFast(claimedState, Date.now() + 2000);
+
+                expect(result).toBeInstanceOf(FastPeerVerificationResult);
+                expect(result.forked).toBeTrue();
+                expect(result.myHeight).toEqual(15);
+                expect(result.hisHeight).toEqual(18);
+                expect(result.highestCommonHeight).toBeUndefined();
```

Comment:
shouldn't this have a value when they have a common block?

Reply:
Actually we don't need to determine highestCommonHeight, because we don't need to determine if peer is forked when its height is higher than ours. We need fork state only when peer height is equal or lower than ours when calculating [network state.](https://github.com/ArkEcosystem/core/blob/81838816f5a4d668fa0cbcf224ad12c05ddd0315/packages/core-p2p/src/network-state.ts#L171-L182) .

That makes Case 1 and Case 2 response actually the same, because we can skip highestCommonHeight calculation to improve performance.